src/com/maddyhome/idea/vim/action/change/insert/InsertAfterLineEndAction.java

```diff
@@ -36,7 +36,7 @@ public InsertAfterLineEndAction() {
 
   private static class Handler extends ChangeEditorActionHandler {
     public boolean execute(@NotNull Editor editor, @NotNull DataContext context, int count, int rawCount, @Nullable Argument argument) {
-      VimPlugin.getChange().insertAfterLineEnd(editor, context);
+      VimPlugin.getChange().insertAfterLineEnd(editor, editor.getCaretModel().getPrimaryCaret(), context);
```

Comment:
Why do you run the handler only for the primary caret? It results in a bug in this example:

```
<caret>foo
<caret>bar
baz
```
when I type `A2`, I expect the result to be:

```
foo2
bar2
baz
```

but it is:

```
foo2
2bar
baz
```

Reply:
This is a temporary kludge (only the primary caret is moved until multicaret `change.insert` actions are not implemented).
I agree that there should have been a TODO note.